From 7ae72162f0fa89db3d174bd3ebf4b1864eff44ec Mon Sep 17 00:00:00 2001
From: Max Bires <jbires@google.com>
Date: Mon, 31 Aug 2020 01:21:43 -0700
Subject: [PATCH 15/15] Key Usage OID should be critical

Fixing a bug in the reference keymaster implementation that put the Key
Usage OID under the non-critical OID sections. This caused
implementations making use of the reference to fail the new OID
filtering tests in CTS.

Bug: 142324725
Test: atest android.keystore.cts.KeyAttestationTest#testEcAttestation
Change-Id: Icfd3ce788fcb799c50230dccbfbb0271dbcc7fb1
---
 km_openssl/attestation_utils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/km_openssl/attestation_utils.cpp b/km_openssl/attestation_utils.cpp
index 1ed523b..12b7a0d 100644
--- a/km_openssl/attestation_utils.cpp
+++ b/km_openssl/attestation_utils.cpp
@@ -253,7 +253,7 @@ keymaster_error_t add_key_usage_extension(const AuthorizationSet& tee_enforced,
 
     X509_EXTENSION_Ptr key_usage_extension(X509_EXTENSION_create_by_NID(nullptr,        //
                                                                         NID_key_usage,  //
-                                                                        false /* critical */,
+                                                                        true /* critical */,
                                                                         key_usage_str.get()));
     if (!key_usage_extension.get()) {
         return TranslateLastOpenSslError();
-- 
2.17.1

